src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/KeyInfo.cs

```diff
@@ -0,0 +1,148 @@
+            return GetXml(xmlDocument);
+        }
+
+        internal XmlElement GetXml(XmlDocument xmlDocument)
+        {
+            // Create the KeyInfo element itself
+            XmlElement keyInfoElement = xmlDocument.CreateElement("KeyInfo", SignedXml.XmlDsigNamespaceUrl);
+            if (!string.IsNullOrEmpty(_id))
+            {
+                keyInfoElement.SetAttribute("Id", _id);
+            }
+
+            // Add all the clauses that go underneath it
+            for (int i = 0; i < _keyInfoClauses.Count; ++i)
+            {
+                XmlElement xmlElement = ((KeyInfoClause)_keyInfoClauses[i]).GetXml(xmlDocument);
+                if (xmlElement != null)
+                {
+                    keyInfoElement.AppendChild(xmlElement);
+                }
+            }
+            return keyInfoElement;
+        }
+
+        public void LoadXml(XmlElement value)
+        {
+            if (value == null)
+                throw new ArgumentNullException("value");
+
+            XmlElement keyInfoElement = value;
+            _id = Utils.GetAttribute(keyInfoElement, "Id", SignedXml.XmlDsigNamespaceUrl);
+
+            XmlNode child = keyInfoElement.FirstChild;
+            while (child != null)
+            {
+                XmlElement elem = child as XmlElement;
+                if (elem != null)
+                {
+                    // Create the right type of KeyInfoClause; we use a combination of the namespace and tag name (local name)
+                    string kicString = elem.NamespaceURI + " " + elem.LocalName;
+                    // Special-case handling for KeyValue -- we have to go one level deeper
+                    if (kicString == "http://www.w3.org/2000/09/xmldsig# KeyValue")
+                    {
+                        XmlNodeList nodeList2 = elem.ChildNodes;
+                        foreach (XmlNode node2 in nodeList2)
+                        {
+                            XmlElement elem2 = node2 as XmlElement;
+                            if (elem2 != null)
+                            {
+                                kicString += "/" + elem2.LocalName;
+                                break;
+                            }
+                        }
+                    }
+                    KeyInfoClause keyInfoClause = (KeyInfoClause)CryptoConfig.CreateFromName(kicString);
+                    // if we don't know what kind of KeyInfoClause we're looking at, use a generic KeyInfoNode:
+                    if (keyInfoClause == null)
+                        keyInfoClause = new KeyInfoNode();
+
+                    // Ask the create clause to fill itself with the corresponding XML
+                    keyInfoClause.LoadXml(elem);
+                    // Add it to our list of KeyInfoClauses
+                    AddClause(keyInfoClause);
+                }
+                child = child.NextSibling;
+            }
+        }
+
+        public Int32 Count
```

Comment:
Converted a handful of Int32->int and Int64->long... Boolean, String and Object previously done